web_src/js/features/common-global.js

```diff
@@ -37,11 +37,10 @@ export function initHeadNavbarContentToggle() {
 }
 
 export function initFootLanguageMenu() {
-  function linkLanguageAction() {
+  async function linkLanguageAction() {
     const $this = $(this);
-    $.get($this.data('url')).always(() => {
-      window.location.reload();
-    });
+    await GET($this.data('url'));
```

Comment:
It's not right to use `GET` here.

Now the language switch doesn't reload the page due to error: `Uncaught (in promise) TypeError: NetworkError when attempting to fetch resource.`

Reply:
Need to wrap it in `try` I suppose to replicate `.always`. Or find a better method that does not rely on throwing at all.

Reply:
Hmm ... it seems to be a misconfiguration problem on gitea.com.

If there is no misconfiguration, there wouldn't be a problem I guess.

The strange thing is that on gitea.com, the first request succeeds to change the language, but the second redirected response make the JS error.

@techknowlogick @lunny 

![image](https://github.com/go-gitea/gitea/assets/2114189/69fe889c-7c89-468d-943f-6a85328c6f22)

Reply:
The root URL of gitea.com does not go to the gitea instance, maybe that's the problem?

Reply:
Thank you for reporting and I will investigate it.